### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by JacintaNgwu**
- [no linter errors](https://github.com/microverseinc/linters-config).✔
- Used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).✔
- Documented the work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).✔
- Followed [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).✔
- Installed [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://reactrouter.com/web/guides/quick-start).

- Used React components.✔

-  Fetched data from the SpaceX API endpoints:✔
Rockets: https://api.spacexdata.com/v4/rockets
Missions: https://api.spacexdata.com/v3/missions
- Dispatched an action to store the selected data in the Redux store:✔
Rockets(id, rocket_name, description, flickr_images)
Missions(mission_id, mission_name, description)
- Used useSelector() Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes✔
- Wrote actions and reducers for booking rockets and joining missions✔
- Wrote actions and reducers for canceling rockets and leaving missions✔
- Switched badges for rocket and mission conditional components✔
- Rendered a list of all reserved rockets and joined missions (using filter()) on the "My profile" page.✔

### CODE REVIEW

File: src/redux/Rocket/rockets.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        id: rocket.id,
+        name: rocket.rocket_name,
+        description: rocket.description,
+        image: rocket.flickr_images[0],
+        reserved: false,
+      };
+      rocketsData.push(singleRocket);
+    });
+    return rocketsData;
+  }
```
Comment (Gambit142) [inline src/redux/Rocket/rockets.js:30]: - [ ] Great work and logic using thunkAPI. the both of you did great 👏 😎 💪

File: src/tests/Myprofile.test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import renderer from 'react-test-renderer';
+import MyProfile from '../components/myprofile';
+import store from '../redux/configurestore';
+
+it('Myprofile renders correctlly', () => {
+  const tree = renderer
+    .create(<Provider store={store}><MyProfile /></Provider>)
+    .toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (Gambit142) [inline src/tests/Myprofile.test.js:12]: - [ ] Kindly use React Testing Library to write tests for your Components. You can use any of these articles as guide when writing tests with React Testing Library 

1. https://www.robinwieruch.de/react-testing-library/
2. https://www.digitalocean.com/community/tutorials/how-to-test-a-react-app-with-jest-and-react-testing-library

React Testing Library is a helper DOM ([Document Object Model](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Introduction)) library that handles interaction with components. 👍 💻

File: public/manifest.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      "src": "logo512.png",
+      "type": "image/png",
+      "sizes": "512x512"
+    }
+  ],
+  "start_url": ".",
+  "display": "standalone",
+  "theme_color": "#000000",
+  "background_color": "#ffffff"
+}
```
Comment (Gambit142) [inline public/manifest.json:25]: - [OPTIONAL] I would advise that you remove all bilerplates tat were created using the `create react app command`. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @JacintaNgwu @ijosue16 👋, 

Great work team!!! This is impressive!!!

## Highlights
- All linters checks passed ✔️ 
- Use of `'@reduxjs/toolkit'` to configure Redux store ✔️ 
- **_Actions,_** **_action creators_** and _**reducers**_ are configured using `duck` pattern ✔️ 
- Has _**Missions**_ page ✔️ 
- Has _**Rockets**_ Page ✔️ 
- Has **_My Profile_** Page ✔️ 
- It can reserve rockets and missions ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- Kindly consider the comment tagged [OPTIONAL] under this review. 💻 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by Kamba56 — APPROVED** [review]
Hi @JacintaNgwu, @ijosue16 , 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 
🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**